packages/lisk-transactions/src/transactions/4_register_multisignature.ts

```diff
@@ -0,0 +1,454 @@
+						tx.id,
+						'.asset.multisignature',
+					),
+			);
+
+		return {
+			id: this.id,
+			status: errors.length === 0 ? Status.OK : Status.FAIL,
+			errors,
+		};
+	}
+
+	public validateSchema(): TransactionResponse {
+		const { status, errors: baseErrors } = super.validateSchema();
+		const valid = validator.validate(
+			multisignatureAssetFormatSchema,
+			this.asset,
+		);
+		const errors = [...baseErrors];
+		const assetErrors = validator.errors
+			? validator.errors.map(
+					error =>
+						new TransactionError(
+							`'${error.dataPath}' ${error.message}`,
+							this.id,
+							error.dataPath,
+						),
+			  )
+			: [];
+
+		errors.push(...assetErrors);
+
+		if (this.type !== TRANSACTION_MULTISIGNATURE_TYPE) {
+			errors.push(new TransactionError('Invalid type', this.id, '.type'));
+		}
+
+		if (
+			this.asset.multisignature.min > this.asset.multisignature.keysgroup.length
+		) {
+			errors.push(
+				new TransactionError(
+					'Invalid multisignature min. Must be less than or equal to keysgroup size',
+					this.id,
+					'.asset.multisignature.min',
+				),
+			);
+		}
+
+		if (!this.amount.eq(0)) {
+			errors.push(
+				new TransactionError(
+					'Amount must be zero for multisignature registration transaction',
+					this.id,
+					'.asset',
+				),
+			);
+		}
+
+		if (this.recipientId) {
+			errors.push(
+				new TransactionError('Invalid recipient', this.id, '.recipientId'),
+			);
+		}
+
+		if (this.recipientPublicKey) {
+			errors.push(
+				new TransactionError(
+					'Invalid recipientPublicKey',
+					this.id,
+					'.recipientPublicKey',
+				),
+			);
+		}
+
+		return {
+			id: this.id,
+			status:
+				status === Status.OK && valid && errors.length === 0
+					? Status.OK
+					: Status.FAIL,
+			errors,
+		};
+	}
+
+	public verify({ sender }: RequiredMultisignatureState): TransactionResponse {
+		const { errors: baseErrors, state } = super.apply({ sender });
+		if (!state) {
+			throw new Error('State is required for applying transaction');
+		}
+		const errors = [...baseErrors];
+
+		// Check if multisignatures already exists on account
+		if (
+			state.sender.multisignatures &&
+			state.sender.multisignatures.length > 0
+		) {
+			errors.push(
+				new TransactionError(
+					'Register multisignature only allowed once per account.',
+					this.id,
+					'.signatures',
+				),
+			);
+		}
+
+		// Check if multisignatures includes sender's own publicKey
+		if (
+			this.asset.multisignature.keysgroup.includes(`+${state.sender.publicKey}`)
+		) {
+			errors.push(
+				new TransactionError(
+					'Invalid multisignature keysgroup. Can not contain sender',
+					this.id,
+					'.signatures',
+				),
+			);
+		}
+
+		return {
+			id: this.id,
+			status: errors.length === 0 ? Status.OK : Status.FAIL,
+			errors,
+		};
+	}
+
+	public apply({ sender }: RequiredMultisignatureState): TransactionResponse {
+		const { errors: baseErrors, state } = super.apply({ sender });
```

Comment:
I don't like the fact that in some transactions, we use the state returned by `super.apply` while in others, we don't. For example, in type 0 transaction, the state returned by `super.apply` is not used. 
What should be the consistent approach?

Reply:
let's create anther issue and consistently use state from the base transaction.